insights.go

```diff
@@ -17,15 +20,27 @@ func unregisterInsights() error {
 }
 
 func insightsIsRegistered() (bool, error) {
+	var errBuffer bytes.Buffer
 	cmd := exec.Command("/usr/bin/insights-client", "--status")
+	cmd.Stderr = &errBuffer
 
 	err := cmd.Run()
 
 	if err != nil {
 		// When the error is ExitError, then we know that insights-client only returned
 		// some error code not equal to zero. We do not care about error number.
-		if _, ok := err.(*exec.ExitError); ok {
-			return false, nil
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			// When stderr is empty, then we know that
+			stdErr := errBuffer.String()
+			if len(stdErr) == 0 {
+				return false, nil
+			} else {
+				if stdErr[len(stdErr)-1] == '\n' {
+					stdErr = stdErr[0 : len(stdErr)-1]
+				}
+				return false, fmt.Errorf("%s", stdErr)
```

Comment:
Couldn't [`strings.TrimSpace`](https://pkg.go.dev/strings#TrimSpace) accomplish this in less code?

```suggestion
				return false, fmt.Errorf("%s", strings.TrimSpace(stdErr))
```